--- /usr/share/perl5/PVE/API2/Nodes.pmbak	2024-06-05 13:59:31.883423670 +0800
+++ /usr/share/perl5/PVE/API2/Nodes.pm	2024-06-05 13:59:31.895423722 +0800
@@ -486,6 +486,12 @@
 	$res->{pveversion} = PVE::pvecfg::package() . "/" .
 	    PVE::pvecfg::version_text();
 
+$res->{sensors_json} = `sensors -j`; # 获取 CPU 、主板温度及风扇转速
+$res->{smartctl_nvme_json} = `smartctl -a -j /dev/nvme?`; # 读取 nvme 硬盘 S.M.A.R.T. 值，获取硬盘寿命、容量、温度等
+$res->{smartctl_sda_json} = `smartctl -i -n standby /dev/sda|grep "STANDBY" || smartctl -i -n standby /dev/sda|grep "No such device" || smartctl -a -j /dev/sda`; #先检测硬盘是否为休眠状态，若否，则检查磁盘是否存在，若否，则返回 S.M.A.R.T. 值
+$res->{smartctl_sdb_json} = `smartctl -i -n standby /dev/sdb|grep "STANDBY" || smartctl -i -n standby /dev/sdb|grep "No such device" || smartctl -a -j /dev/sdb`; #先检测硬盘是否为休眠状态，若否，则检查磁盘是否存在，若否，则返回 S.M.A.R.T. 值
+$res->{cpusensors} = `lscpu | grep MHz`; # 读取 CPU 频率
+
 	my $dinfo = df('/', 1);     # output is bytes
 
 	$res->{rootfs} = {
